src/lib/helpers/BoxHelper.svelte

```diff
@@ -0,0 +1,23 @@
+<script lang="ts">
+    import { type ColorRepresentation, BoxHelper } from 'three'
+    import Object3DInstance from '../instances/Object3DInstance.svelte'
+    import { getParent } from '../internal/HierarchicalObject.svelte'
+  
+    //self
+    export let color:ColorRepresentation = undefined;
+  
+    const parent = getParent()
+  
+    let helper: BoxHelper
+  
+    if (!($parent instanceof Object3DInstance)) {

```

Comment:
`Object3DInstance` is the component, I guess you want to check against `$parent instanceof THREE.Object3D && $parent.type === 'Mesh'`